### ISSUE TITLE
[ENHANCEMENT] Support bundle enhancement

### ISSUE THREAD

**Issue by bk201** [harvester/harvester#3494]
**Is your enhancement related to a problem? Please describe.**

Add more info that might help to debug:
- supportconfig tarball on each node. This contains hardware info and resource usage, which are quite important for troubleshooting.
- harvester settings are not collected. Note we need to check if some settings contain sensitive info and sanitize them.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by ibrokethecloud** [comment]
To test:

* update setting `support-bundle-image` to use the new image `rancher/support-bundle-kit:master-head`
* generate a support bundle
* extract downloaded support bundle

expected result:
* inside the support bundle users should find `$BUNDLEPATH/yamls/cluster/harvesterhci.io/v1beta1/settings.yaml` file which contains extracted settings yamls
* the following settings will be skipped from the file: "cluster-registration-url", "containerd-registry", "additional-ca", "ssl-certificates"
* the nodes zip file `$BUNDLEPATH/nodes/nodeName.zip` when extract will contain a `scc_harvester*.txz` file.

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.2-574927b5-head` (08/17). Close this issue. 

### Result

#### Single node Harvester

- [x] We can find the Harvester settings in `$BUNDLEPATH/yamls/cluster/harvesterhci.io/v1beta1/settings.yaml` 
- [x] We can't find some sensitive settings "cluster-registration-url", "containerd-registry", "additional-ca", "ssl-certificates" in the above files
- [x] We can find the scc_harvester*.txz file under `nodes zip` file in $BUNDLEPATH/nodes/nodeName.zip 

    <details><summary>Test Result $~~$ $\color{green}{\textsf{PASS}}$ </summary><br>
    
     - We can find the Harvester settings in `$BUNDLEPATH/yamls/cluster/harvesterhci.io/v1beta1/settings.yaml` 
        ![image](https://github.com/harvester/harvester/assets/29251855/c74679a3-8303-4206-a136-72f314dacd87)
      
        ![image](https://github.com/harvester/harvester/assets/29251855/3da9e6fa-378c-4368-828d-cd06d38e7b25)
    
     -  We can't find some sensitive settings "cluster-registration-url", "containerd-registry", "additional-ca", "ssl-certificates" in the above files

        ![image](https://github.com/harvester/harvester/assets/29251855/147daf77-d90a-499a-baad-8e908a5421d3)

     - We can find the scc_harvester*.txz file under `nodes zip` file in $BUNDLEPATH/nodes/nodeName.zip 

        ```
        davidtclin@localhost:~/Documents/Harvester/Issues/3494/supportbundle_f9535c50-0540-4604-8fc6-cfd97bf05135_2023-08-22T07-12-06Z/nodes/bp12-230817> find ./ -name *.txz
        ./logs/scc_supportconfig_bp12-230817.txz
        ```
        ![image](https://github.com/harvester/harvester/assets/29251855/3bae2a35-c76e-4aad-937a-a50804fbbfa1)

    </details> 

#### Three nodes Harvester cluster upgrade from v1.1.2

- [x] We can find the Harvester settings in `$BUNDLEPATH/yamls/cluster/harvesterhci.io/v1beta1/settings.yaml` 
- [x] We can't find some sensitive settings "cluster-registration-url", "containerd-registry", "additional-ca", "ssl-certificates" in the above files
- [x] We can find the scc_harvester*.txz file under `nodes zip` file in $BUNDLEPATH/nodes/nodeName.zip 

    <details><summary>Test Result $~~$ $\color{green}{\textsf{PASS}}$ </summary><br>
    
     - We can find the Harvester settings in `$BUNDLEPATH/yamls/cluster/harvesterhci.io/v1beta1/settings.yaml` 
        ![image](https://github.com/harvester/harvester/assets/29251855/e758c1a3-14ca-4130-9f85-e1ff15a99ba7)
          
     -  We can't find some sensitive settings "cluster-registration-url", "containerd-registry", "additional-ca", "ssl-certificates" in the above files

         ![image](https://github.com/harvester/harvester/assets/29251855/242e9b33-01be-4caa-ba33-0ccdf0bd5273)


     - We can find the scc_harvester*.txz file under `nodes zip` file in $BUNDLEPATH/nodes/nodeName.zip 

        ```
        davidtclin@localhost:~/Documents/Harvester/issues/3494/supportbundle_8effddac-94b7-40c2-bb8f-9f632daa0a24_2023-08-22T08-16-55Z/nodes/node3> find ./ -name *.txz
        ./logs/scc_supportconfig_node3.txz
        ```
          ![image](https://github.com/harvester/harvester/assets/29251855/b2113d73-3ca0-485c-9194-8e3384d32564)

    </details> 


### Test Information
* Test Environment: 3 nodes harvester on bare machine and single node on local kvm 
* Harvester version: `v1.2-574927b5-head` (08/17)

### Verify Steps
<details><summary>Execution Steps</summary>

  1. Prepare a fresh install single or three nodes Harvester upgrade from v1.1.2
  2. Open Harvester settings
  3. Set the `support-bundle-image` to use the new image `rancher/support-bundle-kit:master-head`
    ![image](https://github.com/harvester/harvester/assets/29251855/6512fc04-ffae-426a-8276-a32b73581fcb)
  4. Get the support bundle
  5. Extract the support bundle zip
  6. Find the Harvester settings in `$BUNDLEPATH/yamls/cluster/harvesterhci.io/v1beta1/settings.yaml`
  7. Check can't find "cluster-registration-url", "containerd-registry", "additional-ca", "ssl-certificates" in above file
  8. Extract the `$BUNDLEPATH/nodes/nodeName.zip`
  9. Find the `scc_harvester*.txz` under nodeName folder

</details>
